src/algebra/order/complete_field.lean

```diff
@@ -312,3 +312,29 @@ instance : unique (β ≃+*o γ) := unique_of_subsingleton $ induced_order_ring_
 
 end induced_map
 end linear_ordered_field
+
+section real
+
+/-- Any ring homomorphism `ℝ → ℝ` is an ordered ring isomorphism, and thus there are no
+nontrivial ring homorphism `ℝ → ℝ`. -/
+instance real.ring_hom.unique : unique (ℝ →+* ℝ) :=
+{ default := ring_hom.id ℝ,
+  uniq :=
+  begin
+    intro f,
+    have f_mon : monotone f,
+    { intros a b hab,
+      have : f b - f a ≥ 0,
+      { calc
+          f b - f a = f (b - a)                 : (ring_hom.map_sub _ _ _).symm
+            ...     = f (real.sqrt (b - a)^2)   : by rw sq_sqrt (sub_nonneg.mpr hab)
+            ...     = (f (real.sqrt (b - a)))^2 : ring_hom.map_pow _ _ _
+            ...     ≥ 0                         : sq_nonneg _, },
+      linarith, },
+    haveI : subsingleton (ℝ →+*o ℝ):= order_ring_hom.subsingleton,
+    exact congr_arg order_ring_hom.to_ring_hom
+      (subsingleton.elim ⟨f, f_mon⟩ ⟨ring_hom.id ℝ, monotone_id⟩),
+  end,
+}
```

Comment:
I think you can do this as well to compress it to one line:
```suggestion
    exact congr_arg order_ring_hom.to_ring_hom (subsingleton.elim ⟨f, f_mon⟩ ⟨_, monotone_id⟩),
  end }
```

Reply:
Indeed. Thanks!

Reply:
Well, it did not work with giving the function explicitly so I had to cheat a bit to make it fit on one line

Reply:
Oh my bad, Lean can infer the function but not that it's a ring hom ...